cli/injector/grpc_disruption.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package main
+
+import (
+	"strings"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/injector"
+	"github.com/spf13/cobra"
+)
+
+var grpcDisruptionCmd = &cobra.Command{
+	Use:   "grpc-disruption",
+	Short: "GRPC disruption subcommand",
+	Run:   injectAndWait,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		rawEndpointAlterations, _ := cmd.Flags().GetStringArray("endpoint-alterations")
+
+		var endpointAlterations []v1beta1.EndpointAlteration
+
+		// Each value passed to --endpoint-alterations should be of the form `endpoint;alterationtype;alterationvalue`, e.g.
+		// `/chaos_dogfood.ChaosDogfood/order;error;ALREADY_EXISTS`
+		// `/chaos_dogfood.ChaosDogfood/order;override;{}`
+
+		log.Infow("arguments to grpcDisruptionCmd", "endpoint-alterations", rawEndpointAlterations)
+
+		for _, line := range rawEndpointAlterations {
+			split := strings.Split(line, ";")
+			if len(split) != 3 {
+				log.Fatalw("could not parse --endpoint-alterations argument to grpc-disruption", "offending argument", line)
+				continue
+			}
+
+			var endpointAlteration v1beta1.EndpointAlteration
+
+			if split[1] == "error" {
+				endpointAlteration = v1beta1.EndpointAlteration{
+					TargetEndpoint: split[0],
+					ErrorToReturn:  split[2],
+				}
+			} else if split[1] == "override" {
+				endpointAlteration = v1beta1.EndpointAlteration{
+					TargetEndpoint:   split[0],
+					OverrideToReturn: split[2],
+				}
+			} else {
+				log.Fatalw("GRPC injector does not understand alteration type", "type", split[1])
+			}
+
+			endpointAlterations = append(endpointAlterations, endpointAlteration)
+		}
+
+		port, _ := cmd.Flags().GetInt("port")
```

Comment:
Might be easier for a reader if this were co-located at the top with the other `Flags()`